lib/pkg.ml

```diff
@@ -445,13 +445,17 @@ let v ~dry_run
 
 (* Distrib *)
 
-let distrib_version_opam_files ~dry_run p ~version =
-  opam p
-  >>= fun file -> OS.File.read_lines file
-  >>= fun o ->
-  let o = List.filter (fun l -> not (String.is_prefix ~affix:"version" l)) o in
-  let o =  Fmt.strf "version: \"%s\"" version :: o in
-  Sos.write_file ~dry_run file (String.concat ~sep:"\n" o)
+let distrib_version_opam_files ~dry_run _p ~version =
```

Comment:
not sure about this -- the package list can be overriden by the `--packages` option so it's best to call `infer_pkg_names` only in one place in the binary (if `--package` is not set). Could you instead change `p` to be a list of packages and propagate the changes in the rest of the code?